### DISCUSSION TITLE
Search box flickers (GTK 3.24.x and Xorg)

### DISCUSSION THREAD

**Discussion by LinuxOnTheDesktop** [Ulauncher/Ulauncher#911] [General]
### Steps to reproduce
1. Load Ulauncher.
2. Bring up Ulauncher via hotkey.
3. Start to type a search string.
But the problem is only sometimes reproducible, even with the same search string.

### Current behavior

Sometimes, upon step 3, the search box disappears for approximately a second, and then reappears, showing (good) results.

Possibly something has gone wrong with my configuration file.

### Expected behavior

No delay, or at least not a delay in which the search box disappears.

### Environment
**Ulauncher version**: 5.8.0
**Linux distro**: Mint 20 x64
**Desktop environment**: Cinnamon

### Logs
Problematic behaviour was not produced while logging (and I relaunched, with logging, thrice). But the log does show errors.
[Link to log](<script src="https://gist.github.com/LinuxOnTheDesktop/6a26be05b0131c6789fbdf72cf17ac2d.js"></script>).

**Comment by sidouglas** [comment]
Can confirm the same behaviour on PopOS 20.04.

**Comment by troycurtisjr** [comment]
Could this be related to #337? Namely do you happen to be using focus follows mouse or sloppy focus?

**Comment by LinuxOnTheDesktop** [comment]
@troycurtisjr troycurtisjr

Thanks, but I am using neither of those mouse modes.

**Comment by troycurtisjr** [comment]
Now that I'm looking for it I see the same behavior on Fedora 32. It's not long enough that I really noted it as anything other than "It's building the list" type of delay, but there is definitely a delay in the display.

**Comment by gornostal** [comment]
I'm noticing the same issue on pop-os. Started occurring after a recent update.
Assuming I've updated GTK+ which caused this issue, because I haven't seen it before.
My current version is 3.24.20.

After a bit of googling I found this issue reported with the same version of GTK+ https://gitlab.gnome.org/GNOME/gtk/-/issues/2927

The fix has already been merged there, but I don't know which version will have it and when it arrives to debian's repos

**Comment by LinuxOnTheDesktop** [comment]
Re gornostal's comment: I am affected by the problem and I have, I think, GTK 3.24 (just as gornostal has):

```
$ dpkg -l libgtk2.0-0 libgtk-3-0
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name              Version          Architecture Description
+++-=================-================-============-==================================================
ii  libgtk-3-0:amd64  3.24.20-0ubuntu1 amd64        GTK graphical user interface library
ii  libgtk2.0-0:amd64 2.24.32-4ubuntu4 amd64        GTK graphical user interface library - old version
ii  libgtk2.0-0:i386  2.24.32-4ubuntu4 i386         GTK graphical user interface library - old version
```

**Comment by gornostal** [comment]
Looks like the fix is not included in any version of GTK, although it was merged 4 months ago :thinking:

**Comment by PaulVanSchayck** [comment]
Also affected by this (also PopOS! GTK3.24.23-1ubuntu1). The flickers are nearly a second for me, it's really quite annoying. How sure are we this is fixed by the linked pull request in GTK? Did anyone test?

I asked the GTK developers if there is a chance of getting that PR included in a 3.24 point release.

**Comment by PaulVanSchayck** [comment]
See comment on the GTK PR. Apparently the backport to 3.24 isn't trivial. But this fix is included in GTK4. Are we really sure that the thing we're seeing here with Ulauncher is the issue in GTK?

**Comment by gornostal** [comment]
@PaulVanSchayck It may be in a bug in Ulauncher, and some update in GTK just made it so the bug started presenting itself in Ulauncher.
I don't have a solution yet. It's annoying for me too. Although I've started seeing this issue less often in the past month or so.

**Comment by PaulVanSchayck** [comment]
I've switched to Wayland (for other reasons), and this issue disappeared.  

I now have an other issue with some applications not opening through ulauncher though. Always something to tinker with in Linux :P

**Comment by friday** [comment]
I migrated this to Discussions since we can't fix issues like this that are occuring in the GTK code. Hopefully it's fixed by upgrading to GTK4, even though we are importing the GTK3 API compatible version of GTK regardless of being GTK 3 or 4.
